### PR TITLE
Filter ts only keywords from js file completion

### LINKED ISSUE #29212: In JS, completion list suggests TypeScript syntax
**TypeScript Version:**  3.3.0-dev.20181222

**Search Terms:**

**Code**

```ts
// trigger completion list anywhere in a JS file
```

**Expected behavior:**

Completion list doesn't contain TypeScript specific syntax in JS files.

**Actual behavior:**

In JS files it suggests `enum`, `interface`, `namespace`, `type`, `declare`, `abstract`, `readonly`, `public`, `protected`, `private`.

**Playground Link:** 

**Related Issues:**

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #29212

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -1952,6 +1980,40 @@ namespace ts.Completions {
...
+            case SyntaxKind.ModuleKeyword:
+            case SyntaxKind.NamespaceKeyword:
+            case SyntaxKind.NeverKeyword:
+            case SyntaxKind.NumberKeyword:
+            case SyntaxKind.ObjectKeyword:
+            case SyntaxKind.PrivateKeyword:
+            case SyntaxKind.ProtectedKeyword:
+            case SyntaxKind.PublicKeyword:
+            case SyntaxKind.ReadonlyKeyword:
+            case SyntaxKind.StaticKeyword:
```
Comment (weswigham) [inline src/services/completions.ts]: How is `static` TS only?

File: src/services/completions.ts [hunk 2]
```diff
@@ -1952,6 +1980,39 @@ namespace ts.Completions {
...
+        switch (kind) {
+            case SyntaxKind.AbstractKeyword:
+            case SyntaxKind.AnyKeyword:
+            case SyntaxKind.BigIntKeyword:
+            case SyntaxKind.BooleanKeyword:
+            case SyntaxKind.DeclareKeyword:
+            case SyntaxKind.EnumKeyword:
+            case SyntaxKind.GlobalKeyword:
+            case SyntaxKind.InferKeyword:
+            case SyntaxKind.InterfaceKeyword:
```
Comment (ajafff) [inline src/services/completions.ts]: what about `implements`?

**Comment by sheetalkamat** [comment]
Ping! This is 3.4 bug. Thanks

File: src/services/completions.ts [hunk 3]
```diff
@@ -1952,6 +1980,40 @@ namespace ts.Completions {
         }));
     }
 
+    function isTypeScriptOnlyKeyword(kind: SyntaxKind) {
+        switch (kind) {
+            case SyntaxKind.AbstractKeyword:
```
Comment (weswigham) [inline src/services/completions.ts]: Can this list be sourced from a single place, so this and the similar list in `getInJsKeywords` don't drift?  IIRC we have a function for converting from keyword string to kind and another for the reverse.

Comment (sheetalkamat) [reply src/services/completions.ts]: The `getInJsKeywords` is for test and we don't want to mix that but have that list explicit just like any other keyword filter tests that we have.

**Pull request merged**